### ISSUE TITLE
Right prompt support

### ISSUE THREAD

**Issue by DevonVille** [nojhan/liquidprompt#154]
Would be awesome if I can define it via the .liquidpromptrc, and not otherwise hack into the source code.

**Comment by paulvt** [comment]
I too like to have a simple left-prompt that is consistent in size and leave the often much varying in length PWD to be flushed to the right.  (I realize that the liquidprompt philosophy goes a bit against this "consistent prompt size" notion, but it only really varies in size if important things happen such as high loads/temperature or low battery.)

I noticed that it's quite easy to combine both things and have temporarily fixed this up by using a `right-prompt.ps1` file containing a `PS1` definition that leaves out `${LP_PERM}` and `${LP_PWD}`.  Then in my .zshrc file I use (after the sourcing of liquidprompt):

    function precmd {
      _lp_set_prompt
      if [[ -w "${PWD}" ]]; then
          RPS1="${LP_COLOR_WRITE}${LP_PWD}${NO_COL}"
      else
          RPS1="${LP_COLOR_NOWRITE}${LP_PWD}${NO_COL}"
      fi
    }

As a result the right prompt now shows the PWD and it still turns green/red based on whether it's writable.

**Comment by dolmen** [comment]
@paulvt :clap:

**Comment by nojhan** [comment]
Shouldn't such a feature go in a theme, now? The portable compromise seems to be `_lp_fill` (#738), but some themes may probably benefit from having a Zsh variant. Maybe for *Unfold* (#746)?

**Comment by Rycieos** [comment]
I agree with @nojhan, the new `_lp_fill()` functionality is probably good enough to satisfy this requirement. I would like to see some dynamic shifting done, but probably that would have to be done on a theme-by-theme basis, so the Liquidprompt specific work is probably already done, and will be released in v2.2.

**Comment by milouse** [comment]
I also think it can now be set up in theme very easily. Myself I did the following (BTW very inspired by the unfold theme!) and it works very well:

```
_lp_milouse_theme_prompt() {
    _lp_default_theme_prompt_data

    # … blablabla
    
    local left="${LP_BRACKET_OPEN}…blabla"
    local right="${LP_RUNTIME}…blabla"
    local main="${LP_SHLVL}${LP_JOBS}${LP_ERR}${LP_MARK_PREFIX}${LP_COLOR_MARK}${LP_MARK}"

    # shellcheck disable=SC2154
    if (( _LP_SHELL_zsh )); then
        # shellcheck disable=SC2034
        RPS1="$right"
    else
        local lp_fill
        _lp_fill "$left" "$right"
        left="$lp_fill"
    fi

    local n=$'\n'
    PS1="${LP_PS1_PREFIX}${left}${n}${main}${LP_PS1_POSTFIX}"
}
```

Thus maybe we can close this issue?

**Comment by Rycieos** [comment]
@milouse yes, the work that will be released as part of v2.2, specifically [_lp_fill()](https://liquidprompt.readthedocs.io/en/latest/functions/util.html#lp_fill), should be sufficient to close this issue. I'll close this issue when that version is released.

**Comment by milouse** [comment]
Yes sorry I discovered today on my side this feature was not officially released yet 🤦

The perks of running on master ^^
